src/libraries/System.Runtime.InteropServices.JavaScript/src/System/Runtime/InteropServices/JavaScript/JSHostImplementation.cs

```diff
@@ -244,14 +258,48 @@ public static void InstallWebWorkerInterop(bool installJSSynchronizationContext,
 
         public static void UninstallWebWorkerInterop()
         {
-            var ctx = SynchronizationContext.Current as JSSynchronizationContext;
+            var ctx = JSSynchronizationContext.CurrentJSSynchronizationContext;
             var uninstallJSSynchronizationContext = ctx != null;
             if (uninstallJSSynchronizationContext)
             {
+                foreach (var jsObjectWeak in ThreadCsOwnedObjects.Values)
```

Comment:
It may be worth considering what happens here if an object throws during Dispose or one of the TaskCallbacks throws during Invoke. Previously, we would have torn down correctly (since we weren't cleaning up the objects), but now if that happens we won't properly tear down. It might be necessary to put the last bits of cleanup in a `finally` block, and do the dispose/invokes in a `try`.

Reply:
That should not really happen, if it does we are in big trouble and I don't think we should try to recover. I will add abort.

Reply:
I agree that we shouldn't try to recover from it, just make sure shutdown isn't interrupted